builtin/providers/aws/resource_aws_db_parameter_group.go

```diff
@@ -161,18 +161,29 @@ func resourceAwsDbParameterGroupUpdate(d *schema.ResourceData, meta interface{})
 		}
 
 		if len(parameters) > 0 {
-			modifyOpts := rds.ModifyDBParameterGroupInput{
-				DBParameterGroupName: aws.String(d.Get("name").(string)),
-				Parameters:           parameters,
-			}
-
-			log.Printf("[DEBUG] Modify DB Parameter Group: %s", modifyOpts)
-			_, err = rdsconn.ModifyDBParameterGroup(&modifyOpts)
-			if err != nil {
-				return fmt.Errorf("Error modifying DB Parameter Group: %s", err)
+			// We can only modify 20 parameters at a time, so walk them until
+			// we've got them all.
+			maxParams := 20
+			for parameters != nil {
+				paramsToModify := make([]*rds.Parameter, 0)
+				if len(parameters) <= maxParams {
+					paramsToModify, parameters = parameters[:], nil
+				} else {
+					paramsToModify, parameters = parameters[:maxParams], parameters[maxParams:]
+				}
+				modifyOpts := rds.ModifyDBParameterGroupInput{
+					DBParameterGroupName: aws.String(d.Get("name").(string)),
+					Parameters:           paramsToModify,
+				}
+
+				log.Printf("[DEBUG] Modify DB Parameter Group: %s", modifyOpts)
+				_, err = rdsconn.ModifyDBParameterGroup(&modifyOpts)
+				if err != nil {
+					return fmt.Errorf("Error modifying DB Parameter Group: %s", err)
```

Comment:
If this fails part-way through then the state will presumably be inaccurate since it won't be able to tell which parameters have been applied and which haven't. Maybe not a big deal since usually ``terraform refresh`` runs implicitly as part of ``terraform plan`` and that will presumably fix it up...

I'm not sure there's much that can be done about this since the partial state mechanism appears to only work for top-level keys, so it wouldn't work to say ``d.SetPartial("parameter." + parameterName)``. Hopefully this is enough of an edge-case (more than 20 parameters to update *and* an error occurs, and then the user doesn't refresh) that it won't hurt anything. :grinning: